execution/block-partitioner/src/v2/conflicting_txn_tracker.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright © Aptos Foundation
+
+use crate::v2::types::{PreParedTxnIdx, ShardedTxnIndexV2};
+#[cfg(test)]
+use aptos_types::state_store::state_key::StateKey;
+use aptos_types::{
+    block_executor::partitioner::{RoundId, ShardId},
+    transaction::analyzed_transaction::StorageLocation,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::btree_set::BTreeSet;
+
+/// This structure is only used in `V2Partitioner`.
+/// For txns that claimed to access the same storage location,
+/// it caches some metadata about the location and also keeps track of their status (pending or position finalized) throughout the partitioning process.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ConflictingTxnTracker {
+    /// The storage location on which conflicting txns are being tracked by this tracker.
+    pub storage_location: StorageLocation,
+    /// A randomly chosen owner shard of the storage location, for conflict resolution purpose.
+    pub anchor_shard_id: ShardId,
+    /// Txns that (1) read the current storage location and (2) have not been accepted.
+    pending_reads: BTreeSet<PreParedTxnIdx>,
```

Comment:
can we not get rid of `PreParedTxnIdx` type and use `TxnIdx` instead? The name is very confusing and doesn't serve any purpose IMO.